app/javascript/components/RecordingMethodsPreviews/index.test.js

```diff
@@ -0,0 +1,211 @@
+                some other recording method
+              </a>
+            </output>
+          </li>
+        )
+      })
+
+      it('has a sentence with the checked items', () => {
+        expect(recordingMethodsPreviews()).to.contain(
+          <output className="reactive-preview__highlight">
+            voice recording, video recording, and some other recording method
+          </output>
+        )
+      })
+
+      it('defaults to "my child"', () => {
+        expect(recordingMethodsPreviews().text()).to.contain('I understand that my childʼs activities')
+      })
+
+      describe('able to give consent', () => {
+        beforeEach(() => {
+          props.able_to_consent = true
+        })
+
+        it('removes "my child"', () => {
+          expect(recordingMethodsPreviews().text()).to.contain('I understand that my activities')
+        })
+      })
+    })
+  })
+
+  describe('values changing', () => {
+    describe('the researcher name changing', () => {
+      beforeEach(() => {
+        props = {
+          recording_methods: ['voice', 'another_thing'],
+          other_recording_method: 'The other value',
+          all_recording_methods: {
+            'voice': 'the voice',
+            'video': 'the video',
+            'another_thing': 'another thing'
+          }
+        }
+      })
+
+      describe('adding an item that is not there', () => {
+        // Simulate change from a field such as
+        // <input name="research_session[recording_methids][]" data-previewed-by="RecordingMethodsPreviews" />
+        // - usually triggered onchange, but that link is not tested here, we're just simulating the
+        //   resulting event
+        beforeEach(() => {
+          recordingMethodsPreviews().instance().handleCheckboxChange({
+            target: {
+              name: 'research_session[recording_methods][]',
+              value: 'video',
+              checked: true
+            }
+          })
+          recordingMethodsPreviews().update()
+        })
+
+        it('adds that value to the list in order', () => {
+          expect(recordingMethodsPreviews()).to.contain(
+            <ul className="bullet-point-list">
+              <li><output className="reactive-preview__highlight">the voice</output></li>
+              <li><output className="reactive-preview__highlight">the video</output></li>
+              <li><output className="reactive-preview__highlight">another thing</output></li>
+            </ul>
+          )
+        })
+
+        it('adds that value to the sentence in order', () => {
+          expect(recordingMethodsPreviews().text()).to.contain(
+            'the voice, the video, and another thing'
+          )
+        })
+      })
+
+      describe('changing the "other" value', () => {
+        // Simulate change from a field such as
+        // <input name="research_session[recording_methids][]" data-previewed-by="RecordingMethodsPreviews" />
```

Comment:
Tiny one, `recording_methids` is probably meant to be `recording_methods`.

This is repeated in all tests that bypass `onchange`.